data/spells/scripts/party/enchant.lua

```diff
@@ -1,57 +1,65 @@
-local combat = createCombatObject()
-local area = createCombatArea(AREA_CROSS5X5)
-setCombatArea(combat, area)
-setCombatParam(combat, COMBAT_PARAM_EFFECT, CONST_ME_MAGIC_BLUE)
-setCombatParam(combat, COMBAT_PARAM_AGGRESSIVE, 0)
-
-local condition = createConditionObject(CONDITION_ATTRIBUTES)
-setConditionParam(condition, CONDITION_PARAM_SUBID, 3)
-setConditionParam(condition, CONDITION_PARAM_BUFF_SPELL, 1)
-setConditionParam(condition, CONDITION_PARAM_TICKS, 2 * 60 * 1000)
-setConditionParam(condition, CONDITION_PARAM_STAT_MAGICPOINTS, 1)
+local combat = Combat()
+combat:setArea(createCombatArea(AREA_CROSS5X5))
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_MAGIC_BLUE)
+combat:setParameter(COMBAT_PARAM_AGGRESSIVE, 0)
+
+local condition = Condition(CONDITION_ATTRIBUTES)
+condition:setParameter(CONDITION_PARAM_SUBID, 3)
+condition:setParameter(CONDITION_PARAM_BUFF_SPELL, 1)
+condition:setParameter(CONDITION_PARAM_TICKS, 2 * 60 * 1000)
+condition:setParameter(CONDITION_PARAM_STAT_MAGICPOINTS, 1)
 
 local baseMana = 120
-function onCastSpell(cid, var)
-	local pos = getCreaturePosition(cid)
+function onCastSpell(creature, var, isHotkey)
+	local position = creature:getPosition()
 
-	local membersList = getPartyMembers(cid)
-	if(membersList == nil or type(membersList) ~= 'table' or #membersList <= 1) then
-		doPlayerSendCancel(cid, "No party members in range.")
-		doSendMagicEffect(pos, CONST_ME_POFF)
+	local party = creature:getParty()
+	if not party then
+		creature:sendCancelMessage("No party members in range.")
+		position:sendMagicEffect(CONST_ME_POFF)
+		return false
+	end
+
+	local membersList = party:getMembers()
+	membersList = membersList + party:getLeader()
+	if membersList == nil or type(membersList) ~= 'table' or #membersList <= 1 then
+		creature:sendCancelMessage("No party members in range.")
+		position:sendMagicEffect(CONST_ME_POFF)
 		return false
 	end
 
 	local affectedList = {}
-	for _, pid in ipairs(membersList) do
-		if(getDistanceBetween(getCreaturePosition(pid), pos) <= 36) then
-			table.insert(affectedList, pid)
+	for i = 1, #membersList do
+		if getDistanceBetween(membersList[i]:getPosition(), position) <= 36 then
+			affectedList[#affectedList + 1] = membersList[i]
 		end
 	end
 
 	local tmp = #affectedList
-	if(tmp <= 1) then
-		doPlayerSendCancel(cid, "No party members in range.")
-		doSendMagicEffect(pos, CONST_ME_POFF)
+	if tmp <= 1 then
+		creature:sendCancelMessage("No party members in range.")
+		position:sendMagicEffect(CONST_ME_POFF)
 		return false
 	end
 
 	local mana = math.ceil((0.9 ^ (tmp - 1) * baseMana) * tmp)
-	if(getPlayerMana(cid) < mana) then
-		doPlayerSendDefaultCancel(cid, RETURNVALUE_NOTENOUGHMANA)
-		doSendMagicEffect(pos, CONST_ME_POFF)
+	if player:getMana() < mana then
```

Comment:
Tought I fixed all of those, did them abit to quick and copy pasted the spell codes haha..
But it's only this one that I can see needs to be changed.